boards/xiao_m0/examples/eic.rs

```diff
@@ -0,0 +1,112 @@
+//! It uses an external interrupt to toggle an LED.
+//!
+//! You need to connect a button between Pin a9 and ground. Each time the button
+//! is pressed, the LED will toggle his state.
+//!
+//! To select the correct ExtInt
+
+#![no_std]
+#![no_main]
+
+use core::cell::RefCell;
+use cortex_m::{
+    interrupt::{free, Mutex},
+    peripheral::NVIC,
+};
+use hal::{
+    clock::{ClockGenId, GenericClockController},
+    eic::{
+        pin::{ExtInt5, Sense},
+        EIC,
+    },
+    gpio::v2::{Pin, PullUpInterrupt},
+    pac::{self, interrupt, CorePeripherals, Peripherals},
+    prelude::*,
+};
+use panic_halt as _;
+use xiao_m0::{hal, Led0, Led1};
```

Comment:
Can you use the pattern we use elsewhere?
```suggestion
use xiao_m0 as bsp;
use bsp::{hal, Led0, Led1};
```